### ISSUE TITLE
incorrectly `reportUnknownParameterType` with double type alias

### ISSUE THREAD

**Issue by abstract-official** [microsoft/pyright#10776]
## Environment data

-   Pylance version: 2025.7.1
-   OS and version: Windows 11
-   Python version (& distribution if applicable, e.g. Anaconda): python 3.13.5

## Code Snippet

```python
from __future__ import annotations


def f[T](x: a[T]) -> a[T]:
    return x


class C[T]:
    def __init__(self, x: a[T]) -> None:
        self.x = x

    def method(self) -> a[T]:
        return self.x


type a[T] = b[T] | T
type b[T] = C[T]
```

## Repro Steps

1. Paste the above code in an editor in VS Code whose language mode is python, with Pylance enabled (of course).

## Expected behavior

There's no typechecking errors.

## Actual behavior

- The typechecker says 'Type of parameter "x" is unknown' in `C.__init__`, and counts this as an error.
- The typechecker accepts all other usage of `a[T]`, and there's also no other errors reported.
- When hovering the mouse over the parameter "x" in question, the inferred type of "x" is reported to be `a[T@C]`.

## Logs

```
2025-08-08 16:55:56.102 [info] (30432) Pylance language server 2025.7.1 (pyright version 1.1.403, commit 8b02aa84) starting
2025-08-08 16:55:56.122 [info] (30432) Starting service instance "<default>" for workspace ""
2025-08-08 16:55:56.193 [info] (30432) Starting Mcp server id:4140 ...
2025-08-08 16:55:56.247 [info] (30432) Setting environmentName for service "<default>": "3.13.5 (global)"
2025-08-08 16:55:56.248 [info] (30432) Setting pythonPath for service "<default>": "c:\Users\***\AppData\Local\Microsoft\WindowsApps\python3.13.exe"
2025-08-08 16:55:56.248 [info] (30432) No include entries specified; assuming \<default workspace root>
2025-08-08 16:55:56.249 [info] (30432) Auto-excluding **/node_modules
2025-08-08 16:55:56.249 [info] (30432) Auto-excluding **/__pycache__
2025-08-08 16:55:56.250 [info] (30432) Auto-excluding **/.*
2025-08-08 16:55:57.367 [info] (30432) Assuming Python version 3.13.5.final.0
2025-08-08 16:55:58.653 [error] (30432) File or directory "\<default workspace root>" does not exist.
2025-08-08 16:55:58.653 [info] (30432) No source files found.
2025-08-08 16:55:58.718 [info] (30432) BG: Priority queue background worker(2) root directory: file:///c%3A/Users/***/.vscode/extensions/ms-python.vscode-pylance-2025.7.1/dist
2025-08-08 16:55:58.718 [info] (30432) BG: Priority queue background worker(2) started
2025-08-08 16:55:58.718 [info] (30432) Settings have not changed. Skipping update.
```

**Comment by abstract-official** [comment]
The problem disappears when you change `type a[T] = b[T] | T` to `type a[T] = C[T] | T`. In other words, this problems occurs only when dealing with multiple layers of aliases, thus the title.

**Comment by erictraut** [comment]
Thanks for reporting. This looks like a bug in pyright. Could someone from the pylance team please transfer the bug to the pyright issue tracker? Thanks!
